### PR TITLE
Compiler warnings 2

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-07-26-002/index.html

File: Source/driver/Castro.cpp [hunk 1]
```diff
@@ -2345,8 +2345,8 @@ Castro::post_init (Real stop_time)
 
         if (sum_per > 0.0) {
 
-          const int num_per_old = floor((cumtime - dtlev) / sum_per);
-          const int num_per_new = floor((cumtime        ) / sum_per);
+          const int num_per_old = static_cast<int>(std::floor((cumtime - dtlev) / sum_per));
+          const int num_per_new = static_cast<int>(floor((cumtime        ) / sum_per));
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:2349]: These should either both be std::floor or neither, right?

File: Source/driver/Castro.cpp [hunk 2]
```diff
@@ -1820,9 +1820,9 @@ Castro::computeNewDt (int                   finest_level,
 
 void
 Castro::computeInitialDt (int                   finest_level,
-                          int                   sub_cycle,
+                          int                   /*subcycle*/,
                           Vector<int>&           n_cycle,
-                          const Vector<IntVect>& ref_ratio,
+                          const amrex::Vector<amrex::IntVect>& /*ref_ratio*/,
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:1825]: The compiler complained about Vector in this line but not in the previous line?

Comment (zingale) [reply Source/driver/Castro.cpp:1825]: Oh, I probably misunderstood the issue the first time around

Comment (zingale) [reply Source/driver/Castro.cpp]: I removed the amrex::

File: Source/driver/Castro.cpp (hunk 1 above)
Comment (zingale) [reply Source/driver/Castro.cpp]: I added the std::

File: Source/driver/Castro.H [hunk 3]
```diff
@@ -628,33 +595,33 @@ public:
...
 /// @param dt_min
 /// @param dt_level
 /// @param stop_time        End time of simulation
 /// @param post_regrid_flag Have we done regridding yet?
 ///
     void computeNewDt (int                   finest_level,
-                       int                   sub_cycle,
+                       int                   /*subcycle*/,
                        amrex::Vector<int>&           n_cycle,
                        const amrex::Vector<amrex::IntVect>& ref_ratio,
```
Comment (harpolea) [inline Source/driver/Castro.H:626]:
```suggestion
                       const amrex::Vector<amrex::IntVect>& /*ref_ratio*/,
```
The docstring says that this is unused

Comment (zingale) [reply Source/driver/Castro.H]: done

**Pull request merged**